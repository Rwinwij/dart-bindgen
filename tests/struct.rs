use dart_bindgen::{config::*, Codegen};
use insta::assert_snapshot;
use log::LevelFilter;
use simplelog::{TermLogger, TerminalMode};
use std::io::Cursor;

#[test]
fn test_enum() {
    TermLogger::init(
        LevelFilter::Debug,
        Default::default(),
        TerminalMode::Mixed,
    )
    .unwrap();
    let config = DynamicLibraryConfig {
        ios: DynamicLibraryCreationMode::Executable.into(),
        android: DynamicLibraryCreationMode::open("libstruct.so").into(),
        windows: DynamicLibraryCreationMode::open("struct.dll").into(),
        ..Default::default()
    };
    let bindings = Codegen::builder()
        .with_src_header("tests/headers/struct.h")
        .with_lib_name("libstruct")
        .with_config(config)
        .build()
        .unwrap()
        .generate()
        .unwrap();

    let mut out = Cursor::new(Vec::new());
    bindings.write(&mut out).unwrap();
    let out = String::from_utf8(out.into_inner()).unwrap();
    assert_snapshot!(out);
}
