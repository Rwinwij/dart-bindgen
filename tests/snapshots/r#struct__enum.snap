---
source: tests/struct.rs
expression: out
---
/// bindings for `libstruct`

import 'dart:ffi';
import 'dart:io';
import 'package:ffi/ffi.dart' as ffi;

// ignore_for_file: unused_import, camel_case_types, non_constant_identifier_names
final DynamicLibrary _dl = _open();
/// Reference to the Dynamic Library, it should be only used for low-level access
final DynamicLibrary dl = _dl;
DynamicLibrary _open() {
  if (Platform.isWindows) return DynamicLibrary.open('struct.dll');
  if (Platform.isAndroid) return DynamicLibrary.open('libstruct.so');
  if (Platform.isIOS) return DynamicLibrary.executable();
  throw UnsupportedError('This platform is not supported.');
}

/// C struct `NormalStruct`.
class NormalStruct extends Struct {
  
  @Int32()
  int a;
  Pointer<ffi.Utf8> b;
  static Pointer<NormalStruct> allocate() {
    return ffi.allocate<NormalStruct>();
  }


  static NormalStruct from(int ptr) {
    return Pointer<NormalStruct>.fromAddress(ptr).ref;
  }

}

/// C struct `TypedStruct`.
class TypedStruct extends Struct {
  
  @Int32()
  int a;
  @Int32()
  int b;
  Pointer<ffi.Utf8> c;
  @Int64()
  int d;
  static Pointer<TypedStruct> allocate() {
    return ffi.allocate<TypedStruct>();
  }


  static TypedStruct from(int ptr) {
    return Pointer<TypedStruct>.fromAddress(ptr).ref;
  }

}

